tests/lib/screenshot-testing/support/page-renderer.js

```diff
@@ -486,12 +513,26 @@ PageRenderer.prototype._getImageLoadingCount = function () {
 };
 
 PageRenderer.prototype._waitForNextEvent = function (events, callback, i, waitTime) {
+
+    function isEmpty(obj) {
+        for (var key in obj) {
+            if (Object.prototype.hasOwnProperty.call(obj, key)) return false;
+        }
+
+        return true;
+    }
+
     var self = this;
     setTimeout(function () {
-        if (self._getAjaxRequestCount() == 0
-            && self._getImageLoadingCount() == 0
-            && !self._isLoading
-        ) {
+        if (!self._isLoading && !self._isInitializing && !self._isNavigationRequested
+            && (
+                isEmpty(self._resourcesRequested)
+                || (!self._getAjaxRequestCount() && !self._getImageLoadingCount())
+                )
```

Comment:
I can create an inner if condition. I wouldn't assign it to a variable before. We should ideally not having to execute it since eg `_getImageLoadingCount()` is very slow